portal/src/DetailsListWithOrdering.tsx

```diff
@@ -25,36 +26,49 @@ interface OrderColumnButtonsProps {
   index?: number;
   itemCount: number;
   onSwapClicked: (index1: number, index2: number) => void;
+  renderAriaLabel: (index?: number) => string;
 }
 
 const OrderColumnButtons: React.FC<OrderColumnButtonsProps> = function OrderColumnButtons(
   props: OrderColumnButtonsProps
 ) {
+  const { index, itemCount, onSwapClicked, renderAriaLabel } = props;
+  const { renderToString } = React.useContext(Context);
   const onUpClicked = React.useCallback(() => {
-    if (props.index == null) {
+    if (index == null) {
       return;
     }
-    props.onSwapClicked(props.index, props.index - 1);
-  }, [props]);
+    onSwapClicked(index, index - 1);
+  }, [index, onSwapClicked]);
   const onDownClicked = React.useCallback(() => {
-    if (props.index == null) {
+    if (index == null) {
       return;
     }
-    props.onSwapClicked(props.index, props.index + 1);
-  }, [props]);
+    onSwapClicked(index, index + 1);
+  }, [index, onSwapClicked]);
+
+  const ariaLabelUp = React.useMemo(() => {
+    return [renderAriaLabel(index), renderToString("up")].join(" | ");
```

Comment:
I would suggest making it localized. Also, since this label may be used by screen readers, using a complete readable phrase seems better.
Maybe something like "Move {key} up"/"Move {key} down"?

Reply:
Pushed 1 more commit to update the label. The label is `Move {key} up` and {key} is rendered by function provided by user of the component